edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleIdentityLifecycleManager.cs

```diff
@@ -110,12 +111,29 @@ public class ModuleIdentityLifecycleManager : IModuleIdentityLifecycleManager
                 p =>
                 {
                     string connectionString = this.GetModuleConnectionString(p);
-                    return new ModuleIdentity(this.iothubHostName, this.gatewayHostName, this.deviceId, p.Id, new ConnectionStringCredentials(connectionString));
+                    return new ModuleIdentity(
+                        this.iothubHostName,
+                        this.edgeDeviceHostname,
+                        this.GetGatewayHostname(p.Id),
+                        this.deviceId,
+                        p.Id,
+                        new ConnectionStringCredentials(connectionString));
                 });
 
             return moduleIdentities.ToImmutableDictionary(m => ModuleIdentityHelper.GetModuleName(m.ModuleId));
         }
 
+        string GetGatewayHostname(string moduleId)
+        {
+            if (moduleId.Equals(Constants.EdgeAgentModuleIdentityName, StringComparison.OrdinalIgnoreCase) ||
+                moduleId.Equals(Constants.EdgeHubModuleIdentityName, StringComparison.OrdinalIgnoreCase))
+            {
+                return string.Empty;
```

Comment:
Set to empty string as Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleIdentityLifecycleManager.cs is used for docker mode. We will support nested edge if and only if it is iotedged mode